pkg/util/ciutil/codefresh.go

```diff
@@ -0,0 +1,38 @@
+// Copyright 2016-2019, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ciutil
+
+import (
+	"os"
+)
+
+// codefreshCI represents the Codefresh CI system.
+type codefreshCI struct {
+	baseCI
+}
+
+// DetectVars detects the env vars for a Codefresh CI system.
+func (c codefreshCI) DetectVars() Vars {
+	v := Vars{Name: c.Name}
+	v.BuildID = os.Getenv("CF_BUILD_ID")
+	v.BuildType = os.Getenv("CF_BUILD_TRIGGER")
```

Comment:
I am slightly confused by this statement from your PR description.

> I left the build type out since I could not find a proper value to populate it (since many other CIs do not set it, I think there's no issue here).

It appears you are setting the `BuildType` (correctly) to `CF_BUILD_TRIGGER`? According to the docs it looks like that is the correct env var to use as well. 

> Will be an indication of the current build was triggered:
> `build`: The build was triggered from the build button
> `webhook`: The build was triggered from a control version webhook

Reply:
Busy days, my mind is glitching. I could swear I removed this line hahaha.

If the value fits, sounds good. I understood this value was a bit more complex, as if it was a Pull Request or a push to branch. My bad (:

Reply:
> I understood this value was a bit more complex, as if it was a Pull Request or a push to branch. 

Actually, you are right. Looking at the other implementations, `BuildType` is meant to identify if a specific build is a PR or a merge (push) build. So `CF_BUILD_TRIGGER` is in fact in correct in that sense. However, I wonder if we can set it to be either a `PullRequest` or `Push` value based on the value of one of `CF_PULL_REQUEST_*` vars. That is, if one of those vars is set during a PR build, but not during a push build that is enough to distinguish between the two. Do you know what `CF_PULL_REQUEST_ACTION` contains? The docs don't specify what it is clearly.

Reply:
I dumped the env vars in a run of a Pull request, this is what I got (don't worry about the sensitive data, I replaced every value with random stuff):

```bash
export CF_ACCOUNT='account-name'
export CF_API_KEY='3e6bbed4ca50610bd6f11b574329054392c903b4526ee3d2135d92587'
export CF_BRANCH='add-codefresh-ci'
export CF_BRANCH_TAG_NORMALIZED='add-codefresh-ci'
export CF_BRANCH_TAG_NORMALIZED_LOWER_CASE='add-codefresh-ci'
export CF_BRANCH_VERSION_NORMALIZED='add-codefresh-ci'
export CF_BUILD_ID='7987512415e517323270cb85'
export CF_BUILD_INITIATOR='fernandocarletti'
export CF_BUILD_TIMESTAMP='1568840181143'
export CF_BUILD_TRIGGER='webhook'
export CF_BUILD_URL='https://g.codefresh.io//build/7987512415e517323270cb85'
export CF_CLI_RUN_WAIT_FALLBACK='true'
export CF_COMMIT_AUTHOR='fernandocarletti'
export CF_COMMIT_MESSAGE='Add Codefresh CI detection'
export CF_COMMIT_URL='https://github.com/pulumi/pulumi/pull/3247/commits/979eb2b3806fb2a3cea33bf86c7949ce69299966'
export CF_CURRENT_ATTEMPT='1'
export CF_HOST_IP='dind-7987512415e517323270cb85.dind.cf-runtime'
export CF_HOST_NAME='dind-7987512415e517323270cb85.dind.cf-runtime'
export CF_KUBECONFIG_PATH='/codefresh/volume/sensitive/.kube/config'
export CF_LINE_NUMBER_dependencies='0'
export CF_LINE_NUMBER_lint_eslint='0'
export CF_LINE_NUMBER_lint_prettier='0'
export CF_MAX_ATTEMPTS='1'
export CF_PIPELINE_TRIGGER_ID='5d8102a52903429363d0e04c'
export CF_PULL_REQUEST_ACTION='synchronize'
export CF_PULL_REQUEST_ID='314561629'
export CF_PULL_REQUEST_NUMBER='442'
export CF_PULL_REQUEST_TARGET='master'
export CF_REPO_NAME='pulumi'
export CF_REPO_OWNER='credijusto'
export CF_REVISION='979eb2b3806fb2a3cea33bf86c7949ce69299966'
export CF_SHORT_REVISION='979eb2b'
export CF_STEP_NAME='preview'
export CF_URL='https://g.codefresh.io'
export CF_VOLUME_PATH='/codefresh/volume'
export CI='true'
```

I'll do the same for a master branch as soon as I can. I believe the `CF_PULL_REQUEST_*` ones are empty when we are not in a PR build. As soon as I can confirm I'll send a commit for that.

Reply:
I just confirmed, the `CF_PULL_REQUEST_*` are not present when `master` was triggered. I just added a logic to check if the `v.PRNumber` is empty to set the `v.BuildType` as `PullRequest` or `Push`.